--- system/expressionengine/modules/comment/mod.comment (Original).php
+++ system/expressionengine/modules/comment/mod.comment.php
@@ -431,6 +431,17 @@
 			$this->EE->db->order_by($order_by, $this_sort);
 		}
 
+		// -------------------------------------------
+		// 'comment_entries_query' hook.
+		//  - Make changes to the query that grabs the current comments
+		//
+			if ($this->EE->extensions->active_hook('comment_entries_query') === TRUE)
+			{
+				$this->EE->extensions->call('comment_entries_query');
+			}
+		//
+		// -------------------------------------------
+
 		$query = $this->EE->db->get('comments');
 		$result_ids = array();
 
