--- system/expressionengine/modules/comment/mod.comment (Original).php
+++ system/expressionengine/modules/comment/mod.comment.php
@@ -428,12 +428,23 @@
 			
 			$this_sort = ($random) ? 'random' : strtolower($sort);
 
 			$this->EE->db->order_by($order_by, $this_sort);
 		}
 
+		// -------------------------------------------
+		// 'comment_entries_query' hook.
+		//  - Override query that grabs current comments
+		//
+			if ($this->EE->extensions->active_hook('comment_entries_query') === TRUE)
+			{
+				$this->EE->extensions->call('comment_entries_query');
+			}
+		//
+		// -------------------------------------------
+
 		$query = $this->EE->db->get('comments');
 		$result_ids = array();
 
 		if ($query->num_rows() > 0)
 		{
 			foreach ($query->result_array() as $row)
