--- C:\EE Test\ExpressionEngine2Test\system\expressionengine\modules\comment\mod.comment.php	2011-01-19 20:27:03.000000000 -0800
+++ D:\Public\ExpressionEngine2\system\expressionengine\modules\comment\mod.comment.php	2011-01-20 08:35:07.000000000 -0800
@@ -428,23 +428,12 @@
 			
 			$this_sort = ($random) ? 'random' : strtolower($sort);
 
 			$this->EE->db->order_by($order_by, $this_sort);
 		}
 
-		// -------------------------------------------
-		// 'comment_entries_query' hook.
-		//  - Override query that grabs current comments
-		//
-			if ($this->EE->extensions->active_hook('comment_entries_query') === TRUE)
-			{
-				$this->EE->extensions->call('comment_entries_query');
-			}
-		//
-		// -------------------------------------------
-
 		$query = $this->EE->db->get('comments');
 		$result_ids = array();
 
 		if ($query->num_rows() > 0)
 		{
 			foreach ($query->result_array() as $row)
@@ -2637,24 +2626,12 @@
 						'comment_date'	=> $this->EE->localize->now,
 						'ip_address'	=> $this->EE->input->ip_address(),
 						'status'		=> ($comment_moderate == 'y') ? 'p' : 'o',
 						'site_id'		=> $this->EE->config->item('site_id')
 					 );
 
-		// -------------------------------------------
-		// 'comment_entries_insert' hook.
-		//  - Modify any of the soon to be inserted values
-		//
-			if ($this->EE->extensions->active_hook('comment_entries_insert') === TRUE)
-			{
-				$data = $this->EE->extensions->call('comment_entries_insert', $data);
-				if ($this->EE->extensions->end_script === TRUE) return;
-			}
-		//
-		// -------------------------------------------
-
 		$return_link = ( ! stristr($_POST['RET'],'http://') && ! stristr($_POST['RET'],'https://')) ? $this->EE->functions->create_url($_POST['RET']) : $_POST['RET'];
 
 		// Secure Forms check
 		if ($this->EE->security->secure_forms_check($this->EE->input->post('XID')) == FALSE)
 		{
 			$this->EE->functions->redirect(stripslashes($return_link));
